rai/node/wallet.hpp

```diff
@@ -113,6 +114,7 @@ class wallet_store
 	rai::kdf & kdf;
 	rai::mdb_env & environment;
 	MDB_dbi handle;
+	std::recursive_mutex mutex;
```

Comment:
Note: This isn't ideal, and indicative of a bigger restructuring that needs to happen in the wallet API.